flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/util/HiveTableUtil.java

```diff
@@ -175,6 +193,77 @@ public static boolean requireRelyConstraint(byte trait) {
 		return Optional.of(String.join(" and ", filters));
 	}
 
+	/**
+	 * Extract DDL semantics from properties and use it to initiate the table.
+	 */
+	public static void initiateTableFromProperties(Table hiveTable, Map<String, String> properties) {
+		setExternal(hiveTable, properties);
+		setRowFormat(hiveTable.getSd(), properties);
+		setStoredAs(hiveTable.getSd(), properties);
+		setLocation(hiveTable.getSd(), properties);
+	}
+
+	private static void setExternal(Table hiveTable, Map<String, String> properties) {
+		boolean external = Boolean.parseBoolean(properties.remove(TABLE_IS_EXTERNAL));
+		if (external) {
+			hiveTable.setTableType(TableType.EXTERNAL_TABLE.toString());
+			// follow Hive to set this property
+			hiveTable.getParameters().put("EXTERNAL", "TRUE");
+		}
+	}
+
+	private static void setLocation(StorageDescriptor sd, Map<String, String> properties) {
+		String location = properties.remove(TABLE_LOCATION_URI);
+		if (location != null) {
+			sd.setLocation(location);
+		}
+	}
+
+	private static void setRowFormat(StorageDescriptor sd, Map<String, String> properties) {
```

Comment:
A better name? `initiateTableFromProperties` too, I don't know you remove key from properties here...

Reply:
I will mention in the java doc that the keys will be removed. Not sure about the name though. Does `extractRowFormat` sound better? Or any suggested names?

Reply:
`extract` looks better